src/Client.php

```diff
@@ -259,6 +259,28 @@ protected function getForceIpv4Flag(ConnectionInterface $connection)
     }
 
     /**
+     * Extracts the value of the allow-selfsigned option from a given connection.
+     *
+     * @param \Phergie\Irc\ConnectionInterface $connection
+     * @return boolean TRUE to allow self signed certificates, FALSE otherwise
+     */
+    protected function getAllowSelfSignedFlag(ConnectionInterface $connection)
+    {
+        return (boolean) $connection->getOption('allow-selfsigned') ?: false;
+    }
+
+    /**
+     * Extracts the value of the verify_peer option from a given connection.
+     *
+     * @param \Phergie\Irc\ConnectionInterface $connection
+     * @return boolean TRUE to verify peer, FALSE otherwise
+     */
+    protected function getVerifyPeerFlag(ConnectionInterface $connection)
+    {
+        return (boolean) $connection->getOption('allow-verify_peer') ?: false;
```

Comment:
Removed it as I cannot find a proper way of testing this and we might need to refactor socket connections completely due to new insights and possibilities of react socket-client